cpp/include/kvikio/utils.hpp

```diff
@@ -69,6 +71,27 @@ inline constexpr std::size_t page_size = 4096;
   return reinterpret_cast<CUdeviceptr>(devPtr);
 }
 
+/**
+ * @brief Help function to convert value to 64 bit signed integer
+ */
+template <typename T, std::enable_if_t<std::is_integral_v<T>>* = nullptr>
+[[nodiscard]] std::int64_t convert_to_64bit(T value)
+{
+  if (value >= std::numeric_limits<std::int64_t>::max()) {
```

Comment:
This function is probably not performance critical, but we could check if the type can overflow int64 with an `if constexpr`, so some instantiations don't even have the `max()` check.

Reply:
good point, done